ddht/tools/driver/node.py

```diff
@@ -48,29 +57,45 @@ def endpoint(self) -> Endpoint:
     def node_id(self) -> NodeID:
         return self.enr.node_id
 
+    @asynccontextmanager
```

Comment:
Now that this is used twice, it would save some lines for this to be in it's own class, `ListenLock`, which both of these callers can have an instance of.